### PR TITLE
Disable hooks and toolbox by default + remove ClusterRole

### PR DESCRIPTION
**Pull request by AurelienGasser**
A bunch of changes:

- Remove unnecessary rbac capabilities ("resources")
- Change from using a `ClusterRole` to using a `Role`﻿
- rename hook files
- make hooks disabled by default (uninstall chaincode + delete secrets), but enable them in skaffold
- make toolbox disabled by default, but enable it in skaffold

### CODE REVIEW

**Comment by AurelienGasser** [comment]
@Kelvin-M I have two questions

1. **Don't you think it would be better to keep the secret deletion hook enabled by default?** I can't imagine a scenario when you want to `helm delete` the release but keep the secrets. Keeping the secrets is actually dangerous since the next `helm install` won't work (wrong secrets). 

2. **It looks like the chaincode uninstall hook isn't useful**:

- Even with the hook disabled, the chaincode still gets uninstalled (see logs below). 
- **Even when I don't use helm** but I use `kubectl apply` (see [example manifest](https://gist.github.com/AurelienGasser/d1388665590f7fd91bee2bde5a680028)), and then `kubectl delete`, the mycc container disappears. 


```
$ docker ps | grep mycc | wc -l
2
☐ ~/code/hlf-k8s$ helm list
NAME                	REVISION	UPDATED                 	STATUS  	CHART        	APP VERSION	NAMESPACE
network-orderer     	1       	Tue May  5 15:25:41 2020	DEPLOYED	hlf-k8s-1.1.1	           	orderer  
network-org-1-peer-1	1       	Tue May  5 15:25:48 2020	DEPLOYED	hlf-k8s-1.1.1	           	org-1    
network-org-2-peer-1	1       	Tue May  5 15:25:56 2020	DEPLOYED	hlf-k8s-1.1.1	           	org-2    
☐ ~/code/hlf-k8s$ helm get hooks network-org-1-peer-1
☐ ~/code/hlf-k8s$ skaffold delete
Cleaning up...
release "network-orderer" deleted
release "network-org-1-peer-1" deleted
release "network-org-2-peer-1" deleted
☐ ~/code/hlf-k8s$ docker ps | grep mycc | wc -l
0
☐ ~/code/hlf-k8s$
```

**Comment by Kelvin-M** [comment]
@AurelienGasser 
1. I agree with your first point. But as @AlexandrePicosson said, I you just render the manifest and use it with kubectl apply and not helm, the secret deletion jobs will be useless ( but maybe it's not a issue to keep it enable by default).

2.  Chaincode uninstall does two things : 
```
              docker rm -f nid1-{{ $.Release.Name }}-{{ .name }}-{{ .version }} || true
              docker rmi -f $(docker images -q 'nid1-{{ $.Release.Name }}-{{ .name }}-{{ .version }}*') || true
```

The main issue does not comes with the fact that the containers is up (normally it doesn't but it's a sanity first step to remove all containers up). The main issue is the presence of the image in the local docker registry (through the docker daemon). In fact, if you change the chaincode code version, peer will not rebuild the image as is it still present in the local registry. So you can run into issue by having a unwanted (old) version of the chaincode because of this remaining image.

Can you check with `docker images | grep 'mycc'` instead ?

**Comment by AurelienGasser** [comment]
After conversation with @Kelvin-M we decided to enable both hooks by default. because it corresponds to the typical desired behavior.

File: skaffold.yaml [hunk 1]
```diff
@@ -60,6 +60,7 @@ deploy:
           enrollments.creds[1].secret: pwd
           enrollments.creds[1].options: "--id.type orderer"
           enrollments.csrHost: network-orderer.orderer
+          toolbox.enabled: true
```
Comment (ClementGautier) [inline skaffold.yaml:63]: if that's the default behavior you might want to cleanup the skaffold file

Comment (Kelvin-M) [reply skaffold.yaml:63]: Toolbox is disable by default !
```
toolbox:
  enabled: false
```
The skaffold used here suggests a dev setup, so we enable the toolbox :)

Comment (ClementGautier) [reply skaffold.yaml:63]: Oh right, I got it the other way, thx.

**Pull request merged**